db/migrate/20240109180000_fix_unread_comments_inconsistencies.rb

```diff
@@ -10,16 +10,15 @@
 # the unread_comments flag.
 class FixUnreadCommentsInconsistencies < ActiveRecord::Migration[7.0]
   def change
-    num_total_users = 0
+    users = User.all
     num_fixed_users = 0
 
-    User.find_each do |user|
+    users.find_each do |user|
       was_user_fixed = fix_unread_comments_flag(user)
```

Comment:
I think that using `User.all` before `users.find_each` makes the whole point of using `find_each` (working in batches) moot since `User.all` loads all records into memory, see e.g. [here](https://www.mintbit.com/blog/why-to-use-find-each-instead-of-all-in-ruby-on-rails). If you do not want to use an index variable in the loop, you can use `User.count` to get the number of users with one SQL query.

Reply:
Oh yeah, true. I changed it to `User.count`.